query/node.go

```diff
@@ -88,9 +88,10 @@ type functionExpression struct {
 // ---------------
 // These nodes are only present during the parsing step and are not present
 // in the resulting command.
-// There are two types of temporary nodes:
+// There are three types of temporary nodes:
 // * literals (constants in the syntax tree).
 // * lists
+// * evaluation context nodes
```

Comment:
I don't think it's too important, since it's only the logical structure of the code (also, key,values are temporary).